File: web-sources/S8-fwks-palm/breadcrumbs/BreadcrumbsNode.js


import { S8WebFront } from "/S8-pkgs-ui-carbide/S8WebFront.js";
import { S8Object } from "/S8-api/S8Object.js";

import { Popover } from "/S8-pkgs-ui-carbide/popover/Popover.js";




/**
 * 
 */
S8WebFront.CSS_import('/S8-fwks-palm/breadcrumbs/Breadcrumbs.css');


/**
 * 
 */
export class BreadcrumbsNode extends S8Object {

    /** @type {HTMLDivElement} */
    wrapperNode;

    
    /**
     * 
     */
    constructor(){
        super();

        /* <displayed-structure> */
        this.wrapperNode = document.createElement("div");
        this.wrapperNode.classList.add("palm-breadcrumbs-node");

        const slashNode = document.createElement("div");
        slashNode.classList.add("palm-breadcrumbs-node-slash");
        this.wrapperNode.appendChild(slashNode);
        slashNode.innerHTML = "/";

        this.iconNode = document.createElement("div");
        this.iconNode.classList.add("palm-breadcrumbs-node-icon");
        this.wrapperNode.appendChild(this.iconNode);

        this.textNode = document.createElement("div");
        this.textNode.classList.add("palm-breadcrumbs-node-text");
        this.wrapperNode.appendChild(this.textNode);
        /* </displayed-structure> */


        this.tooltipPopover = new Popover();
        this.tooltipPopover.setTheme("dark");
        this.tooltipPopover.setDirection("top");
        this.wrapperNode.appendChild(this.tooltipPopover.getEnvelope());

        this.tooltipContent = document.createElement("div");
        this.tooltipContent.classList.add("palm-breadcrumbs-node-tooltip");
        this.tooltipContent.innerHTML = "This is a tooltip";
        this.tooltipPopover.node.appendChild(this.tooltipContent);

        const _this = this;
        this.tooltipPopover.hide();
        this.wrapperNode.addEventListener("mouseenter", function(){ 
            _this.tooltipPopover.show(); 
        }, false);
        this.wrapperNode.addEventListener("mouseleave", function(){ _this.tooltipPopover.hide(); }, false);
        this.wrapperNode.addEventListener("click", function(){ _this.tooltipPopover.hide(); }, false);
    }


    /**
     * 
     * @returns 
     */
    getEnvelope(){
        return this.wrapperNode;
    }


    /**
    * icon
    * @param {*} code 
    */
    S8_set_icon(code) {
        S8WebFront.SVG_insertByCode(this.iconNode, code, 20, 20);
    }

    /**
     * icon
     * @param {*} code 
     */
    S8_set_text(text) {
        this.textNode.innerHTML = text;
    }


    /** */
    S8_render(){}


    /** */
    S8_dispose(){}
}

File: sources/com/s8/fwks/palm/components/breadcrumbs/BreadcrumbsNode.java
package com.s8.fwks.palm.components.breadcrumbs;

import com.s8.api.web.S8WebFront;
import com.s8.api.web.S8WebObject;
import com.s8.fwks.palm.components.PalmWebSources;
import com.s8.pkgs.ui.carbide.icons.SVG_CarbideIcon;


/**
 * 
 * @author pierreconvert
 *
 */
public class BreadcrumbsNode extends S8WebObject {
	
	
	public static BreadcrumbsNode create(S8WebFront front, SVG_CarbideIcon icon, String text) {
		BreadcrumbsNode node = new BreadcrumbsNode(front);
		node.setIcon(icon);
		node.setText(text);
		return node;
	}
	
	
	
	public final static String WEBPATH = PalmWebSources.ROOT_WEBPATH + "/breadcrumbs/BreadcrumbsNode";
	
	/**
	 * 
	 * @param front
	 */
	public BreadcrumbsNode(S8WebFront front) {
		super(front, WEBPATH);
	}
	
	
	
	
	
	/**
	 * 
	 * @param items
	 */
	public void setIcon(SVG_CarbideIcon icon) {
		vertex.outbound().setUInt16Field("icon", icon.code);
	}
	
	
	/**
	 * 
	 * @param items
	 */
	public void setText(String text) {
		vertex.outbound().setStringUTF8Field("text", text);
	}
	
	
}
